### PR TITLE
Update code in /sample-dapps/evm-token-factory To Support Foundry

### PR DESCRIPTION
**Pull request by ferhatqn**
This update reflects the guide being created around this codebase.

- Add Foundry development folder

### CODE REVIEW

File: sample-dapps/evm-token-factory/README.md [hunk 1]
```diff
@@ -45,7 +52,7 @@ pnpm install
 bun install
 ```
 
-First, run the development server:
+After, start the development server:
 
 ```bash
 npm run dev
```
Comment (0xsergen) [inline sample-dapps/evm-token-factory/README.md:58]: I get the message below after running this command. We may add `@types/node` to the dev dependencies.

```
It looks like you're trying to use TypeScript but do not have the required package(s) installed.
Installing dependencies

If you are not trying to use TypeScript, please remove the tsconfig.json file from your package root (and any TypeScript files in your pages directory).


Installing devDependencies (npm):
- @types/node
```

File: sample-dapps/evm-token-factory/contracts/README.md
Comment (0xsergen) [inline sample-dapps/evm-token-factory/contracts/README.md:1]: 1. The `Architecture` tree should be updated. (`smart_contracts` > `contracts` , `components` and `utils` should be under `app`.
2. Paths that include `smart_contracts` should be updated. (e.g., `smart-contracts/Factory.sol`)

Comment (0xsergen) [reply sample-dapps/evm-token-factory/contracts/README.md:1]: 3. I think commands are missing here > "Ensure [Foundry](https://book.getfoundry.sh/) is installed and navigate inside the contracts directory. Install the required dependencies with the following commands:"

Comment (0xsergen) [reply sample-dapps/evm-token-factory/contracts/README.md:1]: 4. We should update the code, I get this error:  "could not find artifact: `Factory`Did you mean `TokenFactory`?"
```
forge create --rpc-url QUICKNODE_HTTP_URL \
--private-key YOUR_PRIVATE_KEY \
src/Factory.sol:Factory
```

**Review by 0xsergen — COMMENTED** [review]
Worked well, nice UI! 

Left minor suggestions, especially for the README file.

File: sample-dapps/evm-token-factory/contracts/README.md
Comment (0xsergen) [reply sample-dapps/evm-token-factory/contracts/README.md:1]: 5. It's written as 2O, not 20. 
"Click "Create Token" and confirm the transaction to create the ERC-2O token!"

PS: I can't make suggestions for the unchanged content, so I put here all of them.

**Review by amilz — APPROVED** [review]

**Pull request merged**